packages/x-date-pickers-pro/src/internal/hooks/useMultiInputRangeField/useMultiInputDateTimeRangeField.ts

```diff
@@ -0,0 +1,132 @@
+import * as React from 'react';
+import useEventCallback from '@mui/utils/useEventCallback';
+import {
+  unstable_useDateTimeField as useDateTimeField,
+  UseDateTimeFieldComponentProps,
+} from '@mui/x-date-pickers/DateTimeField';
+import {
+  applyDefaultDate,
+  useDefaultDates,
+  useUtils,
+  useValidation,
+} from '@mui/x-date-pickers/internals';
+import { DateRange } from '../../models/range';
+import {
+  UseMultiInputDateTimeRangeFieldDefaultizedProps,
+  UseMultiInputDateTimeRangeFieldParams,
+  UseMultiInputDateTimeRangeFieldProps,
+} from '../../../MultiInputDateTimeRangeField/MultiInputDateTimeRangeField.types';
+import {
+  DateTimeRangeComponentValidationProps,
+  DateTimeRangeValidationError,
+  validateDateTimeRange,
+} from '../validation/useDateTimeRangeValidation';
+import { dateRangePickerValueManager } from '../../../DateRangePicker/shared';
+import { dateTimeRangeFieldValueManager } from '../valueManager/dateTimeRangeValueManager';
+
+export const useDefaultizedDateTimeRangeFieldProps = <TDate, AdditionalProps extends {}>(
+  props: UseMultiInputDateTimeRangeFieldProps<TDate>,
+): UseMultiInputDateTimeRangeFieldDefaultizedProps<TDate> & AdditionalProps => {
+  const utils = useUtils<TDate>();
+  const defaultDates = useDefaultDates<TDate>();
+
+  const ampm = props.ampm ?? utils.is12HourCycleInCurrentLocale();
+  const defaultFormat = ampm
+    ? utils.formats.keyboardDateTime12h
+    : utils.formats.keyboardDateTime24h;
+
+  return {
+    ...props,
+    disablePast: props.disablePast ?? false,
+    disableFuture: props.disableFuture ?? false,
+    format: props.format ?? defaultFormat,
+    minDate: applyDefaultDate(utils, props.minDateTime ?? props.minDate, defaultDates.minDate),
+    maxDate: applyDefaultDate(utils, props.maxDateTime ?? props.maxDate, defaultDates.maxDate),
+    minTime: props.minDateTime ?? props.minTime,
+    maxTime: props.maxDateTime ?? props.maxTime,
+    disableIgnoringDatePartForTimeValidation: Boolean(props.minDateTime || props.maxDateTime),
+  } as any;
```

Comment:
Maybe I'm missing something, but is `AdditionalProps` and this `as any` casting is necessary here?
Wouldn't removing those break something?

```diff
diff --git a/packages/x-date-pickers-pro/src/internal/hooks/useMultiInputRangeField/useMultiInputDateTimeRangeField.ts b/packages/x-date-pickers-pro/src/internal/hooks/useMultiInputRangeField/useMultiInputDateTimeRangeField.ts
index ab5c640c2..bc93340ee 100644
--- a/packages/x-date-pickers-pro/src/internal/hooks/useMultiInputRangeField/useMultiInputDateTimeRangeField.ts
+++ b/packages/x-date-pickers-pro/src/internal/hooks/useMultiInputRangeField/useMultiInputDateTimeRangeField.ts
@@ -24,9 +24,9 @@ import {
 import { dateRangePickerValueManager } from '../../../DateRangePicker/shared';
 import { dateTimeRangeFieldValueManager } from '../valueManager/dateTimeRangeValueManager';
 
-export const useDefaultizedDateTimeRangeFieldProps = <TDate, AdditionalProps extends {}>(
+export const useDefaultizedDateTimeRangeFieldProps = <TDate>(
   props: UseMultiInputDateTimeRangeFieldProps<TDate>,
-): UseMultiInputDateTimeRangeFieldDefaultizedProps<TDate> & AdditionalProps => {
+): UseMultiInputDateTimeRangeFieldDefaultizedProps<TDate> => {
   const utils = useUtils<TDate>();
   const defaultDates = useDefaultDates<TDate>();
 
@@ -45,7 +45,7 @@ export const useDefaultizedDateTimeRangeFieldProps = <TDate, AdditionalProps ext
     minTime: props.minDateTime ?? props.minTime,
     maxTime: props.maxDateTime ?? props.maxTime,
     disableIgnoringDatePartForTimeValidation: Boolean(props.minDateTime || props.maxDateTime),
-  } as any;
+  };
 };
 
 export const useMultiInputDateTimeRangeField = <TDate, TChildProps extends {}>({
@@ -55,7 +55,7 @@ export const useMultiInputDateTimeRangeField = <TDate, TChildProps extends {}>({
   startInputRef,
   endInputRef,
 }: UseMultiInputDateTimeRangeFieldParams<TDate, TChildProps>) => {
-  const sharedProps = useDefaultizedDateTimeRangeFieldProps<TDate, TChildProps>(inSharedProps);
+  const sharedProps = useDefaultizedDateTimeRangeFieldProps<TDate>(inSharedProps);
 
   const { value: valueProp, defaultValue, format, onChange } = sharedProps;
```